src/analyze/analyze.c

```diff
@@ -1802,6 +1804,169 @@ static int dump_syscall_filters(int argc, char *argv[], void *userdata) {
 }
 #endif
 
+#if HAVE_LIBBPF
+static int load_available_kernel_filesystems(Set **ret) {
+        _cleanup_set_free_ Set *filesystems = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        /* Let's read the available filesystems */
+
+        f = fopen("/proc/filesystems", "re");
```

Comment:
you should be able to use `read_virtual_file` instead of re-implementing it

Reply:
`read_virtual_file` doesn't read line by line, which is what I need here. Do you think it's worth it using it and reading the resulting string line by line to get the filesystems?

Reply:
I'd say so, there are annoyances when reading kernel API filesystems that we want to generally abstract away in one place